### ISSUE TITLE
[BUG] Set-PnPMicrosoft365Group fails adding members to a group

### ISSUE THREAD

**Issue by joelfmrodrigues** [pnp/powershell#638]
### Reporting an Issue or Missing Feature
`Set-PnPMicrosoft365Group -Identity $group -Members $members` gives an error when updating group members. Works if updating group owners.

### Expected behavior 
`Set-PnPMicrosoft365Group -Identity $group -Members $members` should update the group with the specified members

### Actual behavior 

Getting following error message when executing the command:

[2021-04-29T13:26:16.058Z] EXCEPTION: Value cannot be null. (Parameter 'source')
[2021-04-29T13:26:16.059Z] 
[2021-04-29T13:26:16.059Z] Exception             : 
[2021-04-29T13:26:16.060Z]     Type       : System.ArgumentNullException
[2021-04-29T13:26:16.060Z]     Message    : Value cannot be null. (Parameter 'source')
[2021-04-29T13:26:16.061Z]     ParamName  : source
[2021-04-29T13:26:16.062Z]     TargetSite : 
[2021-04-29T13:26:16.063Z]         Name          : ThrowArgumentNullException
[2021-04-29T13:26:16.063Z]         DeclaringType : System.Linq.ThrowHelper, System.Linq, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
[2021-04-29T13:26:16.064Z]         MemberType    : Method
[2021-04-29T13:26:16.065Z]         Module        : System.Linq.dll
[2021-04-29T13:26:16.065Z]     StackTrace : 
[2021-04-29T13:26:16.066Z]    at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
[2021-04-29T13:26:16.067Z]    at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
[2021-04-29T13:26:16.068Z]    at PnP.PowerShell.Commands.Utilities.Microsoft365GroupsUtility.UpdateMembersAsync(HttpClient httpClient, Guid groupId, String accessToken, String[] members) in c:\PnPPowerShell\src\Commands\Utilities\Microsoft365GroupsUtility.cs:line 314
[2021-04-29T13:26:16.068Z]    at PnP.PowerShell.Commands.Microsoft365Groups.SetMicrosoft365Group.ExecuteCmdlet() in c:\PnPPowerShell\src\Commands\Microsoft365Groups\SetMicrosoft365Group.cs:line 87
[2021-04-29T13:26:16.069Z]    at PnP.PowerShell.Commands.Base.BasePSCmdlet.ProcessRecord() in c:\PnPPowerShell\src\Commands\Base\BasePSCmdlet.cs:line 56
[2021-04-29T13:26:16.070Z]    at System.Management.Automation.CommandProcessor.ProcessRecord()
[2021-04-29T13:26:16.070Z]     Source     : System.Linq
[2021-04-29T13:26:16.071Z]     HResult    : -2147467261
[2021-04-29T13:26:16.072Z] CategoryInfo          : NotSpecified: (:) [Set-PnPMicrosoft365Group], ArgumentNullException
[2021-04-29T13:26:16.072Z] FullyQualifiedErrorId : System.ArgumentNullException,PnP.PowerShell.Commands.Microsoft365Groups.SetMicrosoft365Group

### Steps to reproduce behavior
Try to add members to a group using `Set-PnPMicrosoft365Group -Identity $group -Members $members`

### What is the version of the Cmdlet module you are running?
1.5.0
Works with 1.3.0 (don't know about versions between)

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
@joelfmrodrigues - Can you check with the latest nightly build ( 1.5.29-nightly )  ? Looks like it is fixed, I tested it and it works

**Comment by joelfmrodrigues** [comment]
@gautamdsheth just tried with 1.5.32-nightly and getting the same error.
Not sure if it could be related, but I am connecting with -AccessToken with Graph scope. Although the same command works when updating owners to probably unrelated

**Comment by erik-larson-bwp** [comment]
I'm receiving the same error when setting Owners.  I am trying to create a new Team.  I want to use New-PnPMicrosoft365Group, but this command is failing to set the owner(s) (though with no error).  I have tried my own login locally as well as a certificate based auth in an Azure Function.

**Comment by joelfmrodrigues** [comment]
Just to provide an update as I was able to get back to this project.
Updated to 1.5.0 and can confirm that the workaround provided by @BipulRaman works - basically using a different command to achieve the same end result

**Comment by gautamdsheth** [comment]
Any chance you can test with the nightly build ? I looked at the code and turns out it is the same used for both commands , so not sure why it is failing for your case.

**Comment by joelfmrodrigues** [comment]
@gautamdsheth managed to test with latest nightly (1.5.54-nightly) and I can confirm that this is now working :) 
Closing the issue as it's fixed for newer versions and there's also a workaround available for previous versions
Many thanks everyone for the comments and help, you rock!

**Comment by argRegEater** [comment]
I ran into similar symptoms, it turned out to be a permissions issue. If you don't have MSGraph's 'User.ReadWrite.All', `New-PnPMicrosoft365Group` will happily create your group but will silently fail on adding '-owners' and 'Members'. It wasn't until I used `Add-PnPMicrosoft365GroupOwner` that it finally flagged the permissions issue. Once I added the permission, `New-PnPMicrosoft365Group` worked as expected.
